src/DotLiquid/Hash.cs

```diff
@@ -24,25 +24,33 @@ public class Hash : IDictionary<string, object>, IDictionary
 
         #region Static construction methods
 
-        public static Hash FromAnonymousObject(object anonymousObject)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="anonymousObject"></param>
+        /// <param name="includeBaseClassProperties">If this is set to true, method will map base class' properties too. </param>
+        /// <returns></returns>
+        public static Hash FromAnonymousObject(object anonymousObject, bool includeBaseClassProperties = false)
         {
             Hash result = new Hash();
             if (anonymousObject != null)
             {
-                FromAnonymousObject(anonymousObject, result);
+                FromAnonymousObject(anonymousObject, result, includeBaseClassProperties);
             }
             return result;
         }
 
-        private static void FromAnonymousObject(object anonymousObject, Hash hash)
+        private static void FromAnonymousObject(object anonymousObject, Hash hash, bool includeBaseClassProperties)
         {
-            Action<object, Hash> mapper = GetObjToDictionaryMapper(anonymousObject.GetType());
+            Action<object, Hash> mapper = GetObjToDictionaryMapper(anonymousObject.GetType(), includeBaseClassProperties);
             mapper.Invoke(anonymousObject, hash);                
         }
 
-        private static Action<object, Hash> GetObjToDictionaryMapper(Type type)
+        private static Action<object, Hash> GetObjToDictionaryMapper(Type type, bool includeBaseClassProperties)
         {
-            if (!mapperCache.TryGetValue(type, out Action<object, Hash> mapper))
+            var cacheKey = type.FullName + (includeBaseClassProperties ? "WithBaseProperties" : "WithoutBaseProperties");

```

Comment:
Can you add a separator between the type name and the suffix?

Reply:
Done